benchmarks/synth-bm/src/contract.rs

```diff
@@ -0,0 +1,169 @@
+    let response_handler_task = tokio::task::spawn(async move {
+        let mut rpc_response_handler = RpcResponseHandler::new(
+            channel_rx,
+            wait_until_channel,
+            ResponseCheckSeverity::Log,
+            num_expected_responses,
+        );
+        rpc_response_handler.handle_all_responses().await;
+    });
+
+    info!("Setup complete, starting to send transactions");
+    let timer = Instant::now();
+    for i in 0..args.num_transactions {
+        let sender_idx = usize::try_from(i).unwrap() % accounts.len();
+        let sender = &accounts[sender_idx];
+
+        let transaction = SignedTransaction::call(
+            sender.nonce,
+            sender.id.clone(),
+            args.receiver_id.clone(),
+            &sender.as_signer(),
+            args.deposit,
+            "sign".to_string(),
+            new_random_mpc_sign_args(&mut rng, args.key_version).to_string().into_bytes(),
+            args.gas,
+            block_service.get_block_hash(),
+        );
+        let request = RpcSendTransactionRequest {
+            signed_transaction: transaction,
+            wait_until: wait_until.clone(),
+        };
+
+        // Let time pass to meet TPS target.
+        interval.tick().await;
+
+        // Await permit before sending the request to make channel buffer size a limit for the
+        // number of outstanding requests.
+        let permit = channel_tx.clone().reserve_owned().await.unwrap();
+        let client = client.clone();
+        tokio::spawn(async move {
+            let res = client.call(request).await;
+            permit.send(res);
+        });
+
+        if i > 0 && i % 200 == 0 {
+            info!("sent {i} transactions in {:.2} seconds", timer.elapsed().as_secs_f64());
+        }
+
+        let sender = accounts.get_mut(sender_idx).unwrap();
+        sender.nonce += 1;
+    }
+
+    info!(
+        "Done sending {} transactions in {:.2} seconds",
+        args.num_transactions,
+        timer.elapsed().as_secs_f64()
+    );
+
+    info!("Awaiting RPC responses");
+    response_handler_task.await.expect("response handler tasks should succeed");
+    info!("Received all RPC responses after {:.2} seconds", timer.elapsed().as_secs_f64());
+
+    info!("Writing updated nonces to {:?}", args.user_data_dir);
+    for account in accounts.iter() {
+        account.write_to_dir(&args.user_data_dir)?;
+    }
+
+    Ok(())
+}
+
+/// Constructs the parameters according to
+/// https://github.com/near/mpc/blob/79ec50759146221e7ad8bb04520f13333b75ca07/chain-signatures/contract/src/lib.rs#L127
+fn new_random_mpc_sign_args(rng: &mut ThreadRng, key_version: u32) -> serde_json::Value {
+    #[derive(Serialize)]
+    struct SignRequest {
+        pub payload: [u8; 32],
+        pub path: String,
+        pub key_version: u32,
+    }
+
+    let mut payload: [u8; 32] = [0; 32];
+    rng.fill(&mut payload);
+    let path = Alphanumeric.sample_string(rng, 16);
```

Comment:
@saketh-are is 16 a reasonable value for the path length? In case the required path length may change, it can be made an argument (like `--key-version` above).